internal/storage/bucket_handle_test.go

```diff
@@ -467,3 +470,372 @@ func (t *BucketHandleTest) TestUpdateObjectMethodWithMissingObject() {
 
 	AssertTrue(errors.As(err, &notfound))
 }
+
+func (t *BucketHandleTest) TestComposeObjectMethodWithDstObjectExist() {
+	// Checking that destination object exist
+	_, err := t.bucketHandle.StatObject(context.Background(),
+		&gcs.StatObjectRequest{
+			Name: TestObjectName,
+		})
+
+	AssertEq(nil, err)
+
+	// Reading content before composing it
+	rc, err := t.bucketHandle.NewReader(context.Background(),
+		&gcs.ReadObjectRequest{
+			Name: TestObjectName,
+			Range: &gcs.ByteRange{
+				Start: uint64(0),
+				Limit: uint64(len(ContentInTestObject)),
+			},
+		})
+
+	AssertEq(nil, err)
+
+	defer rc.Close()
+	dstObjBuf := make([]byte, len(ContentInTestObject))
+	_, err = rc.Read(dstObjBuf)
+
+	AssertEq(nil, err)
+	// Content of destination object, before composing it
+	ExpectEq(ContentInTestObject, string(dstObjBuf[:]))
+
+	// Checking if srcObject exists or not
+	srcObj, err := t.bucketHandle.StatObject(context.Background(),
+		&gcs.StatObjectRequest{
+			Name: TestSubObjectName,
+		})
+
+	AssertEq(nil, err)
+	AssertNe(nil, srcObj)
+
+	// Composing the object
+	composedObj, err := t.bucketHandle.ComposeObjects(context.Background(),
+		&gcs.ComposeObjectsRequest{
+			DstName:                       TestObjectName,
+			DstGenerationPrecondition:     nil,
+			DstMetaGenerationPrecondition: nil,
+			Sources: []gcs.ComposeSource{
+				{
+					Name: TestSubObjectName,
+				},
+			},
+			ContentType: ContentType,
+			Metadata: map[string]string{
+				MetaDataKey: MetaDataValue,
+			},
+			ContentLanguage:    ContentLanguage,
+			ContentEncoding:    ContentEncoding,
+			CacheControl:       CacheControl,
+			ContentDisposition: ContentDisposition,
+			CustomTime:         CustomTime,
+			EventBasedHold:     true,
+			StorageClass:       StorageClass,
+			Acl:                nil,
+		})
+
+	AssertEq(nil, err)
+
+	// Reading content of srcObject
```

Comment:
instead mention validation srcObject to ensure that its not effected.

Reply:
fixed